### PR TITLE
[CM_BE_14] Handling OTP expiry during link

### PR DESCRIPTION
**Pull request by Apoorva-GA**

### CODE REVIEW

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- test/In.ProjectEKA.HipServiceTest/Link/LinkPatientTest.cs  2
- test/In.ProjectEKA.HipServiceTest/Link/LinkPatientControllerTest.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=5394831&bid=17684547)

File: src/In.ProjectEKA.HipService/Link/LinkPatientController.cs [hunk 1]
```diff
@@ -64,11 +64,11 @@ private ActionResult ResponseFrom(ErrorRepresentation errorResponse)
         {
             return errorResponse.Error.Code switch
             {
-                ErrorCode.OtpExpired => (ActionResult) BadRequest(errorResponse),
+                ErrorCode.OtpExpired => (ActionResult) Unauthorized(errorResponse),
                 ErrorCode.MultiplePatientsFound => NotFound(errorResponse),
                 ErrorCode.NoPatientFound => NotFound(errorResponse),
                 ErrorCode.OtpGenerationFailed => StatusCode(StatusCodes.Status500InternalServerError, errorResponse),
-                ErrorCode.OtpInValid => NotFound(errorResponse),
+                ErrorCode.OtpInValid => BadRequest(errorResponse),
```
Comment (ganesanarun) [inline src/In.ProjectEKA.HipService/Link/LinkPatientController.cs:71]: Both cases it's unauthorized only no? at least in the swagger.

File: src/In.ProjectEKA.HipService/Link/LinkPatient.cs [hunk 2]
```diff
@@ -125,7 +125,7 @@ await discoveryRequestRepository.Delete(request.TransactionId, request.Patient.C
             var verifyOtp = await patientVerification.Verify(request.LinkReferenceNumber, request.Token);
             if (verifyOtp != null)
```
Comment (ganesanarun) [inline src/In.ProjectEKA.HipService/Link/LinkPatient.cs:126]: rename the variable name to errorResponse.

File: src/In.ProjectEKA.HipService/Link/LinkPatientController.cs (hunk 1 above)
Comment (Apoorva-GA) [reply src/In.ProjectEKA.HipService/Link/LinkPatientController.cs:71]: Swagger says 400 and 401 both for invalid otp at least for link api. Not really sure what we need to fallow.

Comment (ganesanarun) [reply src/In.ProjectEKA.HipService/Link/LinkPatientController.cs:71]: https://stackoverflow.com/questions/30078459/what-should-be-http-status-code-for-credentials-expired-error-exception

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- test/In.ProjectEKA.HipServiceTest/Link/LinkPatientTest.cs  2
- test/In.ProjectEKA.HipServiceTest/Link/LinkPatientControllerTest.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=5394831&bid=17684547)

**Pull request merged**